File: 2020/1.kt
// Advent of code 2020 day 1
import java.io.File

fun main() {
  val fileName = "input1.txt"
  val lines: Set<Int> = File(fileName).readLines().map { it.toInt() }.toSet()

  for (i in lines) {
    for (j in lines) {
      val remainder = 2020 - (i + j)
      if (lines.contains(remainder)) {
        println("Result: " + i * j * remainder)
        return
      }
    }
  }
}

File: 2020/1.py
#!/usr/bin/env python

with open("input1.txt", "r") as f:
  lines = [int(x.strip()) for x in f.readlines()]

for i in lines:
  for j in lines:
    if i + j == 2020:
      print i * j

for i in lines:
  for j in lines:
    for k in lines:
      if i + j + k == 2020:
        print i, j, k
        print i * j * k


